src/react-styleguide/src/components/Inputs/Input.js

```diff
@@ -0,0 +1,220 @@
+import React from 'react'
+import { SuccessIcon, PlusIcon, MinusIcon,
+         XIcon, BangIcon, QuestionIcon, IconExamples } from '../Icons/Icons'
+
+const commonClasses = 'db w-100 pl3 pr5 pv3 lh-copy br2 bg-dark-navy ba eh-text-field'
+const standardClasses = 'white b--gray focus-b-gray'
+const errorClasses = 'red b--red focus-b--red'
+const errMsgClases = 'db red f5 sans-serif mt3'
+const successClasses = 'white b--green focus-b--green'
+const disabledClasses = 'disabled'
+
+const Input = ({type='text', placeholder, required=false, error=false,
+                errMsg, disabled=false, value, icon}) => {
+  let inputStyles = standardClasses
+
+  // If field is required, set style to error or success
+  if (required) {
+    inputStyles = error ? errorClasses : successClasses
+  }
+
+  return (
+    <div className='relative eh-text-field-wrapper'>
+      <input type={type} placeholder={placeholder} required={required} disabled={disabled} defaultValue={value}
+        className={`${commonClasses} ${inputStyles} ${disabled ? disabledClasses : ''}`}
+      />
+      {icon}
+      {errMsg
+        ? <div className={errMsgClases}>{errMsg}</div>
+        : null 
+      }
+    </div>
+  )
+}
+
+Input.propTypes = {
+  type: React.PropTypes.string,
```

Comment:
Would be good to be explicit about what types are accepted with a `oneOf(types)` like this: https://github.com/eggheadio/egghead-instructor-center/blob/master/src/App/screens/Instructor/components/Icon/index.js#L39